core/src/main/java/com/linecorp/armeria/client/WebClient.java

```diff
@@ -314,6 +315,19 @@ default HttpResponse get(String path) {
         return execute(RequestHeaders.of(HttpMethod.GET, path));
     }
 
+    /**
+     * Sends an HTTP GET request by appending the provided query params to the path.
+     */
+    @CheckReturnValue
+    default HttpResponse get(String path, QueryParams params) {
+        requireNonNull(params, "params");
+        if (params.isEmpty()) {
+            return this.get(path);
+        }
+
+        return execute(RequestHeaders.of(HttpMethod.GET, String.format("%s?%s", path, params.toQueryString())));
```

Comment:
- If a path contains `?` already, we should concatenate a query string with `&`.
- The performance of `String.format()` rather slow than `+` or `StringBuilder`. Could you use `+` for merging two strings?
- It would be convenient to support `QueryParams` in other APIs such as `post()`, `patch()` and so on.

Reply:
I've pushed the mentioned changes, had a small doubt about code coverage: should I add tests for each of the overloaded methods that have QueryParams as a parameter? Writing those tests seems redundant to me as long as we're testing the core method appending the QueryParams string and integration with at least one method.

Reply:
It might be tiresome to test all overloaded methods. Could you add integration tests for `GET` and `POST`, please?

Reply:
I've added tests for the `GET` and `POST` methods of WebClient, although I'm not entirely sure if this is what you meant. Please let me know if I'm mistaken.